stdlib/socket/from.go

```diff
@@ -136,20 +136,25 @@ func createFromSocketSource(s plan.ProcedureSpec, dsid execute.DatasetID, a exec
 	// known issue with url.Parse for detecting the presence of a scheme: https://github.com/golang/go/issues/19779
 	var scheme, address string
 	if !strings.Contains(spec.URL, "://") {
-		// no scheme specified, use default and use the entire url as address
-		scheme = schemes[0]
-		address = spec.URL
-	} else {
-		// scheme specified, use appropriate values
-		url, err := neturl.Parse(spec.URL)
-		if err != nil {
-			return nil, err
-		}
-		scheme = url.Scheme
-		address = url.Host
-		if !contains(schemes, scheme) {
-			return nil, errors.Newf(codes.Invalid, "invalid scheme %s, must be one of %v", scheme, schemes)
-		}
+		// no scheme specified, use default scheme and use the entire url as address
```

Comment:
`URLValidator` needs to take a `*url.URL` parameter. So I just tweaked the logic here a little bit to let both branches converge to the validation phase.